routes/api/authUtils.js

```diff
@@ -3,43 +3,38 @@ var keystone = require('keystone')
 var leaderAPIKey = process.env.LEADER_API_KEY;
 
 module.exports = {
+   signin: function(req, res) {
+      if (!req.body.username || !req.body.password) return res.json({ success: false });
 
-	signin: function(req, res) {
-  
-		if (!req.body.username || !req.body.password) return res.json({ success: false });
-  
-		keystone.list('User').model.findOne({ email: req.body.username }).exec(function(err, user) {
-			if (err || !user) {
-				return res.json({
-					success: false,
-						message: (err && err.message ? err.message : false) || 'Sorry, there was an issue signing you in, please try again.'
-				});
-			}
+      keystone.list('User').model.findOne({ email: req.body.username }).exec(function(err, user) {
+         if (err || !user) {
+            return res.json({
+               success: false,
+               message: (err && err.message ? err.message : false) || 'Sorry, there was an issue signing you in, please try again.'
+            });
+         }
 
-			keystone.session.signin({ email: user.email, password: req.body.password }, req, res, function(user) {
-                		if (req.body.gcmId) {
-                    			var model = keystone.list("User").model;
-                    			user.getUpdateHandler(req).process({email:user.email, gcmId:req.body.gcmId}, function(err) {
-                        			console.log(err)
-                    			})
-                		}
-				return res.json({
-					success: true,
-					LeaderAPIKey: leaderAPIKey 
-				});
-			}, function(err) {
-				return res.json({
-					success: false,
-					message: (err && err.message ? err.message : false) || 'Sorry, there was an issue signing you in, please try again.'
-				});
-			});
-		});
-	},
-
-	signout: function(req, res) {
-		keystone.session.signout(req, res, function() {
-			res.json({ 'signedout': true });
-		});
-	}
+         keystone.session.signin({ email: user.email, password: req.body.password }, req, res, function(user) {
+            if (req.body.gcmId) {
+               user.set({ gcmId: req.body.gcmId });
```

Comment:
This is where the bug was happening in the old version. Not sure if this is how you do it.